percona/controller/pgcluster/backup.go

```diff
@@ -0,0 +1,151 @@
+package pgcluster
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	batchv1 "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/percona/percona-postgresql-operator/internal/naming"
+	v2 "github.com/percona/percona-postgresql-operator/pkg/apis/pgv2.percona.com/v2"
+)
+
+func (r *PGClusterReconciler) reconcileBackupJobs(ctx context.Context, cr *v2.PerconaPGCluster) error {
+	for _, repo := range cr.Spec.Backups.PGBackRest.Repos {
+		backupJobs, err := listBackupJobs(ctx, r.Client, cr, repo.Name)
+		if err != nil {
+			return errors.Wrap(err, "failed to list backup jobs")
+		}
+
+		for _, job := range backupJobs.Items {
+			if err := reconcileBackupJob(ctx, r.Client, cr, job, repo.Name); err != nil {
+				return errors.Wrap(err, "failed to reconcile backup job")
+			}
+		}
+	}
+	return nil
+}
+
+func reconcileBackupJob(ctx context.Context, cl client.Client, cr *v2.PerconaPGCluster, job batchv1.Job, repoName string) error {
```

Comment:
Could we cover this with integration test maybe,  at least the happy path?

Reply:
https://github.com/percona/percona-postgresql-operator/pull/524/commits/414b5dc7e48ed8ab6a67e32c29abc8a29b6d2c1f